rhombus/private/array.rkt

```diff
@@ -167,33 +168,54 @@
            body
            dest-id)])))
 
-(define/arity (copy v)
+(define (check-array who v)
+  (unless (vector? v)
+    (raise-argument-error* who rhombus-realm "Array" v)))
+
+(define (check-nonneg-int who n)
+  (unless (exact-nonnegative-integer? n)
+    (raise-argument-error* who rhombus-realm "NonnegInt" n)))
+
+(define/arity (Array.get v i)
```

Comment:
A drawback of defining this `Array.get` with extra checks is that it prevents inlining of `vector-ref` to implement array access. The alternative is reflected by "runtime-config.rkt": use `vector-ref` directly, but rewrite error messages. The current "runtime-config.rkt" doesn't have a mapping from `vector-ref` to Array.get`, but the idea was to fill all those in, eventually.